compiler/resolution/callDestructors.cpp

```diff
@@ -1887,6 +1912,45 @@ static void removeElidedOnBlocks() {
   }
 }
 
+static void insertAutoDestroyPrimsForLoopExprTemps() {
+  // can we cache the CallExprs we care about in resolveCall etc?
```

Comment:
This code seems very work-around-y. Could you add a comment that says it's a workaround and a better solution would be preferred ? (At the moment, I don't know what the better solution would be... but I think it helps to clearly mark things that are workarounds).